### ISSUE TITLE
Unhide subs if loaded track is forced

### ISSUE THREAD

**Issue by Ghosthree3** [mpv-player/mpv#4271]
If the sub track loaded by default by the player has the Forced flag enabled then if the user also has --no-sub-visibility set in their config that option should be reversed to --sub-visibility as the Forced flag is usually used for sub tracks with "non english parts" only. If the user has subs hidden by default they may not be aware this track even exists and miss out on dialog that the director intended to be seen in foreign speaking parts.

I say 'loaded by default' because if your --slang preference is to another language that's available it will still be loaded by default, ignoring the fact there's a track asking to be Forced. Unless that language also has a track with a Forced flag, then it'll use that obviously. (EDIT: I'm not saying change this, this is good, just the thread title).

EDIT: It's said better than I have here, http://kodi.wiki/view/subtitles#Using_Forced_Subtitles
Basically I'm asking if mpv could do like Kodi does for Forced subs.

I guess the title I should have used was "Make mpv handle Forced subs in the same way Kodi does".

**Comment by wm4** [comment]
That seems very weird...

**Comment by Identifier** [comment]
I'm also trying to accomplish this, for exactly the reason you said in which I want to hide subs by default unless they're for scenes with foreign-spoken parts that the director intended to be seen (presumably by including forced subtitles).  

I would think that you could accomplish this by setting --sub-forced-only instead of --no-sub-visibility (i.e., leave sub visibility on but only forced subtitles should appear).  But, even when I set --sub-forced-only, I still see subtitles on .MKV files by default even though the subtitles are not marked as forced (neither on the tracks nor the individual subtitles within them, according to mkvinfo and mkvextract).  This makes me revert to setting --no-sub-visibility globally in my mpv.conf file and potentially missing out on foreign-spoken dialog for which I'm not even aware subtitles should have been shown, like you said.

I realize this thread is over a year old, but I'm curious if others are having problems with --sub-forced-only like I am which might lead them to this feature request you suggested.  Or I'm interested in how this feature request would differ from what --sub-forced-only is already supposed to do.

**Comment by Dudemanguy** [comment]
Definitely don't think we'll be adding an option to directly do this. But the desired behavior should be possible with the correct combination of settings + autoprofiles or a script (didn't attempt to verify myself but oh well). Basically just check if `sub-forced-only-cur` is true and then flip sub-visibility on.

**Comment by Identifier** [comment]
I don't think anyone wanted an option to do this.  The issue is that the existing option(s) aren't working properly.

**Comment by Dudemanguy** [comment]
`--no-sub-visibility` not showing any subs is correct. Making that secretly show subtitles based on certain conditions is not really intuitive. Like I said, I'm pretty sure the desired behavior is easily achievable with a few lines. If not, I'll be happy to reopen this.

**Comment by Identifier** [comment]
The --no-sub-visibility was only needed because of the --sub-forced-only flag not having any effect when slang was specified.

Granted, I haven't checked to see if this is still a problem since 2018 since I personally switched to mpv.net a long time ago.
